drivers/kubevirt/kubevirt.go

```diff
@@ -507,3 +515,350 @@ func AddPVCsToVirtualMachine(vm kubevirtv1.VirtualMachine, pvcs []*corev1.Persis
+		return "", false, nil
+	}
+	_, err = task.DoRetryWithTimeout(t, defaultVmMountCheckTimeout, defaultVmMountCheckRetryInterval)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetVirtLauncherPodForVM returns the virt-launcher pod for the VM
+func GetVirtLauncherPodForVM(virtualMachineCtx *scheduler.Context, vol *volume.Volume) (*corev1.Pod, error) {
+	pods, err := core.Instance().GetPodsUsingPV(vol.ID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get pods for volume %s of context %s: %w", vol.ID, virtualMachineCtx.App.Key, err)
+	}
+
+	var found corev1.Pod
+	for _, pod := range pods {
+		if pod.Labels["kubevirt.io"] == "virt-launcher" && pod.Status.Phase == corev1.PodRunning {
+			if found.Name != "" {
+				// there should be only one VM pod in the running state (otherwise live migration is in progress)
+				return nil, fmt.Errorf("more than 1 KubeVirt pods (%s, %s) are in running state for volume %s",
+					found.Name, pod.Name, vol.ID)
+			}
+			found = pod
+		}
+	}
+	if found.Name == "" {
+		return nil, fmt.Errorf("failed to find a running pod for volume %s", vol.ID)
+	}
+	return &found, nil
+}
+
+// IsVMBindMounted checks if the volumes are bind mounted to the VM
+func IsVMBindMounted(virtualMachineCtx *scheduler.Context, driver scheduler.Driver, volumeDriver volume.Driver, initialCheck bool) (bool, error) {
+	vols, err := driver.GetVolumes(virtualMachineCtx)
+	if err != nil {
+		return false, err
+	}
+
+	// Get the node where pod is scheduled
+	vms, err := GetAllVMsFromScheduledContexts([]*scheduler.Context{virtualMachineCtx})
+	if err != nil {
+		return false, err
+	}
+	// TODO Add support for multiple vm per spec/Context : https://purestorage.atlassian.net/browse/PTX-23167
+	if len(vms) != 1 {
+		return false, fmt.Errorf("expected 1 VM for VM [%s] in namespace [%s] but got [%d]", virtualMachineCtx.App.Key, virtualMachineCtx.App.NameSpace, len(vms))
+	}
+
+	vm := vms[0]
+	vmNodeName, err := GetNodeOfVM(vm)
+	if err != nil {
+		return false, err
+	}
+	log.Infof("VM [%s] is deployed on node [%s]", virtualMachineCtx.App.Key, vmNodeName)
+
+	// Keep a track of replicaset and consider this as the source of truth
+	volInspect, err := volumeDriver.InspectVolume(vols[0].ID)
+	if err != nil {
+		return false, fmt.Errorf("failed to inspect volume [%s]: %w", vols[0].ID, err)
+	}
+	globalReplicSet := volInspect.ReplicaSets
+	log.InfoD("Length of replicaset: %d", len(globalReplicSet))
+
+	// The criteria to call the bind mount successful is to check if the replicaset of all volumes should be same and should be locally attached to the node
+	// check if the replicaset values is same as the global replicaset
+	// Here we are considering globalreplicaset to be source of truth for comparison
+
+	for _, vol := range vols {
+		err := IsVolumeBindMounted(volumeDriver, virtualMachineCtx, vmNodeName, vol, initialCheck)
+		if err != nil {
+			return false, err
+		}
+		err = VerifyVolumeReplicaCollocation(vol, globalReplicSet, volumeDriver)
```

Comment:
VerifyVolumeReplicaCollocation -> AreVolumeReplicasCollacated to be consistent with the naming style of IsVolumeBindMounted

Reply:
Done.